### PR TITLE
Add platform support matrix

### PR DESCRIPTION
**Pull request by mikewiebe**
##### SUMMARY
Update nxos platform documentation to include a support matrix.

We often get questions about which platforms we have certified the nxos modules against.  This matrix provides a single pointer that we can share.

**NOTE:** Not sure if this is the best place for it but I could not find any other locations that looked suitable. 

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
nxos modules

### CODE REVIEW

File: docs/docsite/rst/network/user_guide/platform_nxos.rst [hunk 1]
```diff
@@ -122,3 +122,37 @@ Example NX-API Task
 In this example the ``proxy_env`` variable defined in ``group_vars`` gets passed to the ``environment`` option of the module used in the task.
 
 .. include:: shared_snippets/SSH_warning.txt
+
+***********************************
```
Comment (samccann) [inline docs/docsite/rst/network/user_guide/platform_nxos.rst]:
```suggestion
```

File: docs/docsite/rst/network/user_guide/platform_nxos.rst [hunk 2]
```diff
@@ -122,3 +122,37 @@ Example NX-API Task
...
+Cisco Nexus Platform Support Matrix
+***********************************
```
Comment (samccann) [inline docs/docsite/rst/network/user_guide/platform_nxos.rst]:
```suggestion
===================================
```

File: docs/docsite/rst/network/user_guide/platform_nxos.rst [hunk 3]
```diff
@@ -122,3 +122,37 @@ Example NX-API Task
...
+
+The following platforms and software versions have been certified to work with this version of Ansible.
```
Comment (samccann) [inline docs/docsite/rst/network/user_guide/platform_nxos.rst]: Is this RH certified or Cisco-certified for these platforms and versions?

**Review by samccann — CHANGES_REQUESTED** [review]
Looks good but a couple of questions, especially around what is meant by 'certified' here?

File: docs/docsite/rst/network/user_guide/platform_nxos.rst (hunk 3 above)
Comment (mikewiebe) [reply docs/docsite/rst/network/user_guide/platform_nxos.rst]: This is mostly a statement about Cisco certification but they are also certified/validated by RH.

Our Cisco team runs all of the `nxos_*` tests in `https://github.com/ansible/ansible/tree/devel/test/integration/targets` against our platform matrix.

Comment (gundalow) [reply docs/docsite/rst/network/user_guide/platform_nxos.rst]: Maybe something like:
```suggestion
The following platforms and software versions have been certified by Cisco to work with this version of Ansible.
```

**Review by samccann — APPROVED** [review]

**Comment by samccann** [comment]
Thanks @mikewiebe for the Ansible documentation addition!

**Pull request merged**